hack/verify-receipts-upgrade-migration.sh

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script tests "krew system receipts-migration" which was introduced for
+# migrating krew 0.2.x to krew 0.3.x.
+#
+# TODO(ahmmetb,corneliusweig) remove at/after krew 0.4.x when receipts-migration
+# is no longer supported.
+
+set -euo pipefail
+
+[[ -n "${DEBUG:-}" ]] && set -x
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+BINDIR="${SCRIPTDIR}/../out/bin"
+goos="$(go env GOOS)"
+goarch="$(go env GOARCH)"
+
+install_krew_0_2_1() {
+    # from https://github.com/kubernetes-sigs/krew/blob/v0.2.1/README.md
+    krew_root="${1}"
+    cd "$(mktemp -d)"
```

Comment:
Can you do `pushd "$(mktemp -d)"` and `popd` instead?

Also, it would be nicer to clean up the temporary directory as you do it below
```
trap 'rm -rf "${temp_dir}"' RETURN
```

Reply:
pushd/popd shouldn't really be necessary since "cd" is function-scoped.
adding the trap.

Reply:
If the function is called in a sub-shell, `cd` wont have global effects. But here, `cd` is not confined to the function. For example, this should be technically the same:
```bash
#!/bin/bash

ls

do_sth() {
	echo "do_sth called"
	cd /
	ls
}

do_sth
ls
```
Here, the third `ls` prints the same as the second (with bash-4).

It doesn't really matter for the correctness of this script, but it's still something to consider.